File: packages/aws-cdk/test/commands/test-resources/stacks/S3Stack.cs
using Amazon.CDK;
using Amazon.CDK.AWS.S3;
using Constructs;
using System.Collections.Generic;

namespace GoodCSharpStack
{
    public class GoodCSharpStackProps : StackProps
    {
    }

    /// <summary>
    /// AWS CloudFormation Sample Template S3_Website_Bucket_With_Retain_On_Delete: Sample template showing how to create a publicly accessible S3 bucket configured for website access with a deletion policy of retain on delete.
    /// </summary>
    public class GoodCSharpStack : Stack
    {
        /// <summary>
        /// URL for website hosted on S3
        /// </summary>
        public object WebsiteURL { get; } 

        /// <summary>
        /// Name of S3 bucket to hold website content
        /// </summary>
        public object S3BucketSecureURL { get; } 

        public GoodCSharpStack(Construct scope, string id, GoodCSharpStackProps props = null) : base(scope, id, props)
        {

            // Resources
            var s3Bucket = new CfnBucket(this, "S3Bucket", new CfnBucketProps
            {
                AccessControl = "PublicRead",
                WebsiteConfiguration = new CfnBucket.WebsiteConfigurationProperty
                {
                    IndexDocument = "index.html",
                    ErrorDocument = "error.html",
                },
            });

            // Outputs
            WebsiteURL = s3Bucket.AttrWebsiteURL;
            S3BucketSecureURL = string.Join("", new []
            {
                "https://",
                s3Bucket.AttrDomainName,
            });
        }
    }
}

File: packages/aws-cdk/test/commands/test-resources/stacks/S3Stack.java
package com.myorg;

import software.constructs.Construct;

import java.util.*;
import software.amazon.awscdk.CfnMapping;
import software.amazon.awscdk.CfnTag;
import software.amazon.awscdk.Stack;
import software.amazon.awscdk.StackProps;

import software.amazon.awscdk.*;
import software.amazon.awscdk.services.s3.*;

class GoodJavaStack extends Stack {
    private Object websiteUrl;

    private Object s3BucketSecureUrl;

    public Object getWebsiteUrl() {
        return this.websiteUrl;
    }

    public Object getS3BucketSecureUrl() {
        return this.s3BucketSecureUrl;
    }

    public GoodJavaStack(final Construct scope, final String id) {
        super(scope, id, null);
    }

    public GoodJavaStack(final Construct scope, final String id, final StackProps props) {
        super(scope, id, props);

        CfnBucket s3Bucket = CfnBucket.Builder.create(this, "S3Bucket")
                .accessControl("PublicRead")
                .websiteConfiguration(CfnBucket.WebsiteConfigurationProperty.builder()
                        .indexDocument("index.html")
                        .errorDocument("error.html")
                        .build())
                .build();

        s3Bucket.applyRemovalPolicy(RemovalPolicy.RETAIN);

        this.websiteUrl = s3Bucket.getAttrWebsiteUrl();
        CfnOutput.Builder.create(this, "CfnOutputWebsiteURL")
                .key("WebsiteURL")
                .value(this.websiteUrl.toString())
                .description("URL for website hosted on S3")
                .build();

        this.s3BucketSecureUrl = String.join("",
                "https://",
                s3Bucket.getAttrDomainName());
        CfnOutput.Builder.create(this, "CfnOutputS3BucketSecureURL")
                .key("S3BucketSecureURL")
                .value(this.s3BucketSecureUrl.toString())
                .description("Name of S3 bucket to hold website content")
                .build();

    }
}
